databus/src/main/java/com/bazaarvoice/emodb/databus/db/cql/CqlSubscriptionDAO.java

```diff
@@ -52,15 +56,17 @@ public CqlSubscriptionDAO(CassandraKeyspace keyspace, Clock clock) {
     @Override
     public void insertSubscription(String ownerId, String subscription, Condition tableFilter, Duration subscriptionTtl, Duration eventTtl) {
         insertSubscription(new DefaultOwnedSubscription(
-                subscription,
-                tableFilter,
-                new Date(_clock.millis() + subscriptionTtl.toMillis()),
-                Duration.ofSeconds(Ttls.toSeconds(eventTtl, 1, Integer.MAX_VALUE)),
-                ownerId)
+                        subscription,
+                        tableFilter,
+                        new Date(_clock.millis() + subscriptionTtl.toMillis()),
+                        Duration.ofSeconds(Ttls.toSeconds(eventTtl, 1, SUBSCRIPTION_TTL_LIMIT)),
+                        ownerId
+                ),
+                Ttls.toSeconds(subscriptionTtl, 1, SUBSCRIPTION_TTL_LIMIT)
         );
     }
 
-    private void insertSubscription(OwnedSubscription subscription) {
+    private void insertSubscription(OwnedSubscription subscription, int ttl) {
         Map<String, Object> json = new HashMap<String, Object>() {{
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Move the contents of this initializer to a standard constructor or to field initializers. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS1171)
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use another way to initialize this instance. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarqube.qa.bazaarvoice.com/coding_rules#rule_key=squid%3AS3599)